core/app/models/spree/store_credit.rb

```diff
@@ -23,10 +23,12 @@ class Spree::StoreCredit < Spree::PaymentSource
   validates_numericality_of :amount_used, { greater_than_or_equal_to: 0 }
   validate :amount_used_less_than_or_equal_to_amount
   validate :amount_authorized_less_than_or_equal_to_amount
+  validates :expires_at, presence: true, unless: ->(credit) { credit.category.non_expiring? }
```

Comment:
To be honest, I'm not sure what the intention was with the original implementation. Since there was never an expiry date set anywhere, all credits were essentially `non_expiring`, though some had the `expiring` type.

In my mind when I read that a store credit has an `expiring` type, I would expect it to have an expiry date. This does make existing `expiring` credits invalid, so I understand the argument for maintaining compatibility with current behaviour.